### ISSUE TITLE
c++ nn:FractionalMaxPool2d/3d output_ratio option is integer, should be double

### ISSUE THREAD

**Issue by meganset** [pytorch/pytorch#33240]
## 🐛 Bug

The fractional max pooling options for output ratio need to be an expanding array of doubles.
currently:

```
template <size_t D>
struct FractionalMaxPoolOptions {
  FractionalMaxPoolOptions(ExpandingArray<D> kernel_size)
      : kernel_size_(kernel_size) {}

  /// the size of the window to take a max over
  TORCH_ARG(ExpandingArray<D>, kernel_size);

  /// the target output size of the image
  TORCH_ARG(c10::optional<ExpandingArray<D>>, output_size) = c10::nullopt;

  /// If one wants to have an output size as a ratio of the input size, this option can be given.
  /// This has to be a number or tuple in the range (0, 1)
  TORCH_ARG(c10::optional<ExpandingArray<D>>, output_ratio) = c10::nullopt;
```

later, in nn/functional/pooling.h, if no output sizes, output ratios are used:
```
  if (output_size_ == c10::nullopt) {
    TORCH_INTERNAL_ASSERT(output_ratio != c10::nullopt);
    output_size_ = {(int64_t)(input.sizes()[2] * (*output_ratio.value())[0]),
                    (int64_t)(input.sizes()[3] * (*output_ratio.value())[1]),
                    (int64_t)(input.sizes()[4] * (*output_ratio.value())[2])};
  }
```

the code compiles with integers, but won't allow for output ratios of 0.5, for example

cc @yf225

**Comment by yf225** [comment]
@meganset Thanks a lot for catching this bug! Curious would you like to open a PR to fix it?

**Comment by meganset** [comment]
Fixed with PR #33304
